src/nrniv/netpar.cpp

```diff
@@ -1285,6 +1292,19 @@ void BBS::netpar_solve(double tstop) {
 	tstopunset;
 }
 
+void nrnthread_all_spike_vectors_return(std::vector<double>& spiketvec, std::vector<int>& spikegidvec){
+    if(all_spiketvec != nullptr && all_spikegidvec != nullptr) {
+        all_spiketvec->resize(spiketvec.size());
+        all_spikegidvec->resize(spikegidvec.size());
+        assert(all_spiketvec->capacity() == all_spikegidvec->capacity());
+        for(int i = 0; i < all_spiketvec->capacity(); ++i ) {
+            all_spiketvec->elem(i) = spiketvec[i];
+            all_spikegidvec->elem(i) = spikegidvec[i];
+        }
```

Comment:
again, I don't know the code but my undemanding was that: we will iterate every `spikegidvec` and then deliver event to respective `PreSyn` object with the same GID. This way the implementation work if PreSyns are using underlying different vectors.

I didn't understand fully how global `all_spiketvec` solution works here.

Reply:
I'm not sure we need to deliver events, since we are in "record" mode? 

On `spike_record`, either we have `gid=-1` for all gids or a `gidlist`, which would normally target a subset of all gids?
The deal with different vectors is not as straightforward. Say we record spikes for a subset of gids. On spikes return from CORENEURON we would have all gids spike data. We can match underlying different vectors, but then we need to consider writing out.dat  for the gids that were not recorded in NEURON.

Reply:
We did discuss the idea of being generic with respect to the ways NEURON can have different vectors for recording spikes (and not even necessarily use gids but a user defined identifier). I don't remember if spikes are recorded on the CoreNEURON side when gid < 0 (the negative number can potentially identify a PreSyn without a gid) and just discarded if out.dat is written. But I'm not averse to the direct implementation Alex provided.  It just may mean that a NEURON spike raster may have or be missing spike trains depending on whether CoreNEURON did the simulation or NEURON did the simulation. But I am ok with only supporting the NEURON case ```pc.spike_record(-1)```.  Might  be a good idea to comment about this limited support in the implementation.

We also discussed about a callback not being needed and just use a direct call into NEURON. But I realize not that would be a problem if CoreNEURON did not have NEURON in memory (eg. file transfer mode).

Reply:
Ok so discussed things through with Pramod. Keeping `pc.spike_record(-1)` code as optimisation (and apparently that would the regular case), and added code to handle `pc.spike_record(gidlist)`.

Reply:
Furthermore, we also cover the case of using `pc.spike_record(-1)` and then `pc.spike_record(gidlist)` by invalidating `all_*vec` pointers